### DISCUSSION TITLE
Proxmox CPU Scaling Governor

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#94] [General]
![Screenshot 2022-03-25 2 36 00 PM](https://user-images.githubusercontent.com/21342670/160181542-6a96715f-c8bb-495c-914f-2d23e9ce984e.png)

**Comment by davidmerrique** [comment]
Love this.

Looks like this could be persisted by adding this line to crontab 

```
@reboot echo "ondemand" | tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor
```

via https://www.reddit.com/r/homelab/comments/bltm26/comment/emul6ek

**Reply by semool** [reply]
With "cpufrequtils" you can set this in a config file "/etc/default/cpufrequtils". Example: GOVERNOR="performance"

**Reply by afentoe** [reply]
Yes, it would be a nice addition to the script, to be able to save the settings after a reboot.
Setting /etc/default/cpufrequtils requires the package cpufrequtils to be installed, but is the easiest way to persist the CPU governor setting.
I think it's easy  to implement: You can just overwrite the file if the script is run again.

Another option is to add a line to /etc/crontab:
@reboot  root  sleep 30; echo "powersave" | tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor >/dev/null 2>&1

But this is more difficult to parse/modify, when the script is run multiple times.

**Comment by elmakus** [comment]
when I proceed with Y after the first question I have error:
cat: /sys/class/net/eno1/address: No such file or directory

**Comment by tteck** [comment]
What is the name of your network device?

![Screenshot 2022-03-27 3 05 05 AM](https://user-images.githubusercontent.com/21342670/160270686-7dd8b0bd-0d18-41b9-bb28-24ce34402a70.png)
?

**Reply by elmakus** [reply]
enp7s0f0

**Comment by tteck** [comment]
@elmakus I removed it, try the script now

**Reply by elmakus** [reply]
perfect. thanks!
